arch/arm/include/aarch32/cortex_m/tz.h

```diff
@@ -306,6 +307,46 @@ typedef void __attribute__((cmse_nonsecure_call)) (*tz_ns_func_ptr_t) (void);
 
 #endif /* CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS */
 
+/**
+ * @brief Name of a non-secure entry function if it is to be wrapped.
+ */
+#define __TZ_NSC_NAME(name) nsc_ ## name
+
+/**
+ * @brief Create a thread safe wrapper function for an NS entry function.
+ *
+ * This locks the scheduler before calling the function by wrapping the NS entry
+ * function in @ref k_sched_lock / @ref k_sched_unlock, using
+ * @ref Z_ARM_WRAP_FUNC.
+ *
+ * In secure code:
+ *
+ *	__TZ_NONSECURE_ENTRY_FUNC int __TZ_NSC_NAME(foo)(char *arg)
+ *	{ ... }
+ *
+ * In non-secure code:
+ *
+ *	int foo(char *arg);
```

Comment:
Does the Non-Secure firmware really declare the function?

Reply:
Yes, though it will usually be in a header file.